src/snowflake/cli/api/sql_execution.py

```diff
@@ -98,22 +98,15 @@ def use(self, object_type: ObjectType, name: str):
             )
 
     def current_role(self) -> str:
-        *_, cursor = self._execute_string(
-            "select current_role()", cursor_class=DictCursor
-        )
-        role_result = cursor.fetchone()
-        return role_result["CURRENT_ROLE()"]
+        return self._execute_query(f"select current_role()").fetchone()[0]
 
     @contextmanager
     def use_role(self, new_role: str):
         """
         Switches to a different role for a while, then switches back.
         This is a no-op if the requested role is already active.
         """
-        role_result = self._execute_query(
```

Comment:
`_execute_query` ends up calling `_execute_string` anyways, so the behaviour here doesn't change